dashboard/public/pixelation/pixelation.js

```diff
@@ -80,7 +80,14 @@ function initProjects() {
         options.projectData = levelSource;
       },
       getLevelSource: function () {
-        return pixel_data.value.replace(/[ \n]/g, "");
+        // Store the source in whichever format the level specifies.
+        if (isHexSelected()) {
+          var hexCode = pixel_data.value.replace(/[^0-9A-F]/gi, "");
+          return isHexLevel() ? hexCode : hexToBinPvt(hexCode);
+        } else {
+          var binCode = pixel_data.value.replace(/[^01]/gi, "");
+          return isHexLevel() ? binToHexPvt(binCode) : binCode;
+        }
```

Comment:
The only invalid characters we expect are spaces or newlines. We try to keep other characters out of the data, but it's possible that other bad characters could somehow get in there, perhaps if someone somehow put hex code into the browser when binary was expected. I copied these regexes from elsewhere in pixelation.js. Let me know if you think it's worth extracting.